### PR TITLE
Created a script to run tito builds and upack leftover artifacts

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

replaces https://github.com/openshift/origin/pull/11244
.cc @smarterclayton

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -239,13 +239,22 @@ install-travis:
...
 build-rpms-redistributable:
-	tito build --test --rpm --no-cleanup --rpmbuild-options='--define "make_redistributable 1"'
+	hack/build-rpm-release.sh
 .PHONY: build-rpms-redistributable
+
+# Build a release of OpenShift using tito for linux/amd64 and the images that depend on it.
+#
+# Example:
+#   make release
+rpm-release: clean build-rpms
```
Comment (smarterclayton) [inline Makefile]: release-rpm

Comment (smarterclayton) [inline Makefile]: Actually release-rpms

File: hack/build-rpm-release.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# This script generates release zips and RPMs into _output/releases.
+# tito and other build dependencies are required on the host. We will
+# be running `hack/build-cross.sh` under the covers, so we transitively
+# consume all of the relevant envars.
+source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
+os::build::setup_env
+os::util::environment::setup_tmpdir_vars "build-rpm-release"
+
+if [[ "${OS_ONLY_BUILD_PLATFORMS:-}" == 'linux/amd64' ]]; then
```
Comment (smarterclayton) [inline hack/build-rpm-release.sh]: What's the difference between make_redistributable?  If I ran brew/tito on linux/386, what happens?

Comment (stevekuznetsov) [inline hack/build-rpm-release.sh]: If it's set to `1`, we build all of the clients. If not, we only build the `linux/amd64` one.

https://github.com/openshift/origin/blob/master/origin.spec#L200-L206

File: Makefile (hunk 1 above)
Comment (stevekuznetsov) [inline Makefile]: So what's your take on this? You'll get RPMs and I imagine you could email them to someone ... `release-rpms`?

File: hack/build-rpm-release.sh (hunk 2 above)
Comment (smarterclayton) [inline hack/build-rpm-release.sh]: I would expect make_redistributable to build the platform I am on.

File: Makefile (hunk 1 above)
Comment (smarterclayton) [inline Makefile]: `release-rpms` is good for now.

File: hack/build-rpm-release.sh (hunk 2 above)
Comment (smarterclayton) [inline hack/build-rpm-release.sh]: It's fine for now.  I don't think it makes a ton of sense in the future,  but can live with it.

**Comment by stevekuznetsov** [comment]
@danmcp please re-tag

**Comment by smarterclayton** [comment]
[merge]

**Comment by smarterclayton** [comment]
[merge]

**Comment by stevekuznetsov** [comment]
@smarterclayton this went into `master` yesterday...

File: hack/build-rpm-release.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	os::log::error 'No _output artifact directory found in tito rpmbuild artifacts!'
+	exit 1
+fi
+
+# clean up our local state so we can unpack the tito artifacts cleanly
+make clean
+
+# migrate the tito artifacts to the Origin directory
+mkdir -p "${OS_OUTPUT}"
+cp -r "${tito_output_directory}"/* "${OS_OUTPUT}"
```
Comment (smarterclayton) [inline hack/build-rpm-release.sh]: why not `mv`?  Creates extra content we don't need probably.

File: hack/build-rpm-release.sh [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+mkdir -p "${OS_LOCAL_RELEASEPATH}"
+cp "${tito_tmp_dir}"/x86_64/*.rpm "${OS_LOCAL_RELEASEPATH}"
```
Comment (smarterclayton) [inline hack/build-rpm-release.sh]: Not sure how I feel about this.  On the one hand everything is in the same spot.  On the other, we'll eventually have multi arch.  Hrm.

File: hack/build-rpm-release.sh [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+if [[ "${OS_ONLY_BUILD_PLATFORMS:-}" == 'linux/amd64' ]]; then
+	# when the user is asking for only Linux binaries, we will
+	# furthermore not build cross-platform clients in tito
+	make_redistributable=0
+else
+	make_redistributable=1
+fi
+
+os::log::info 'Building Origin release RPMs with tito...'
+tito_tmp_dir="${BASETMPDIR}/tito"
```
Comment (stevekuznetsov) [inline hack/build-rpm-release.sh]: Outputs a bunch of other nonsense we don't want to keep around. I copy the important bits back at the end.

File: hack/build-rpm-release.sh (hunk 3 above)
Comment (stevekuznetsov) [inline hack/build-rpm-release.sh]: Good question, should be a `mv`

**Pull request merged**